addon/mixins/ajax-request.js

```diff
@@ -225,15 +225,20 @@ export default Mixin.create({
    */
   request(url, options) {
     const hash = this.options(url, options);
-    return new Promise((resolve, reject) => {
-      this._makeRequest(hash)
-        .then(({ response }) => {
-          resolve(response);
-        })
-        .catch(({ response }) => {
-          reject(response);
-        });
+    const internalPromise = this._makeRequest(hash);
+
+    const ajaxPromise = new AJAXPromise((resolve, reject) => {
+      internalPromise.then(({ response }) => {
+        resolve(response);
+      })
+      .catch(({ response }) => {
+        reject(response);
+      });
     }, `ember-ajax: ${hash.type} ${hash.url} response`);
```

Comment:
I think we should leave this be for this PR and consider simplifying things here as a separate matter.